rust/lanzatool/src/generation.rs

```diff
@@ -1,47 +1,71 @@
+use serde::{Deserialize, Serialize};
+
 use std::fmt;
 use std::fs;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 
 use anyhow::{Context, Result};
+use bootspec::BootJson;
+use bootspec::SpecialisationName;
+use bootspec::generation::Generation;
+
+// TODO: actually, I'm not sure it's a good thing to have Default
+// we should maybe have TryDefault?
+// discuss this with upstream.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct SecureBootExtension {
+    #[serde(rename="org.lanzaboote.osRelease")]
+    pub os_release: PathBuf
+}
 
-use crate::bootspec::Bootspec;
+pub type ExtendedBootJson = BootJson<SecureBootExtension>;
 
 #[derive(Debug)]
-pub struct Generation {
+pub struct OSGeneration {
+    /// Top-level nixpkgs version
     version: u64,
-    specialisation_name: Option<String>,
-    pub bootspec: Bootspec,
+    /// Top-level specialisation name
+    specialisation_name: Option<SpecialisationName>,
+    /// Top-level bootspec document
+    pub bootspec: ExtendedBootJson,
+}
+
+fn into_boot_json(generation: Generation<SecureBootExtension>) -> Result<ExtendedBootJson> {
```

Comment:
We could move this into the bootspec module and use a newtype. Then we keep the clearer separation of bootspec logic and generation logic. Not a dealbraker though.

Reply:
Can add it to the upstream PR to DetSys Bootspec, this is a good point actually.